templates/common/upgrade/upgrade.go

```diff
@@ -125,42 +129,90 @@ type Params struct {
 	Version string
 }
 
-type ResultType string
+// ResultType is the outcome of attempting an upgrade. It may succeed, may be
+// unnecessary, or may have some sort of merge conflict.
+type ResultType int
 
 const (
 	// This is the value of Result.Type when the upgrade was vacuously
 	// successful because the template is already on the latest version.
-	AlreadyUpToDate ResultType = "already_up_to_date"
+	AlreadyUpToDate ResultType = iota
```

Comment:
Curious, what's usgae of iota here?

Reply:
It's a Go feature, https://go.dev/ref/spec#Iota